src/services/goToDefinition.ts

```diff
@@ -233,20 +233,24 @@ namespace ts.GoToDefinition {
     }
 
     function getDefinitionFromSymbol(typeChecker: TypeChecker, symbol: Symbol, node: Node): DefinitionInfo[] | undefined {
-        return getConstructSignatureDefinition() || getCallSignatureDefinition() || map(symbol.declarations, declaration => createDefinitionInfo(declaration, typeChecker, symbol, node));
+        // There are cases when you extend a function by adding properties to it afterwards,
+        // we want to strip those extra properties
+        const filteredDeclarations = symbol.declarations.filter(d =>  !ts.isAssignmentDeclaration(d) || d === symbol.valueDeclaration)

```

Comment:
@orta what are these checker changes for? unrelated?

Reply:
Ah, that is another PR of mine, will rebase it out